packages/core/src/fields/types/bigInt/index.ts

```diff
@@ -30,67 +29,87 @@ export type BigIntFieldConfig<ListTypeInfo extends BaseListTypeInfo> =
 
-    const isNullable = getResolvedIsNullable(_validation, config.db)
-
     if (hasAutoIncDefault) {
       if (meta.provider === 'sqlite' || meta.provider === 'mysql') {
-        throw new Error(`The bigInt field at ${meta.listKey}.${meta.fieldKey} specifies defaultValue: { kind: 'autoincrement' }, this is not supported on ${meta.provider}`)
+        throw new Error(`${meta.listKey}.${meta.fieldKey} specifies defaultValue: { kind: 'autoincrement' }, this is not supported on ${meta.provider}`)
       }
+      const isNullable = resolveDbNullable(validation, config.db)
       if (isNullable !== false) {
         throw new Error(
-          `The bigInt field at ${meta.listKey}.${meta.fieldKey} specifies defaultValue: { kind: 'autoincrement' } but doesn't specify db.isNullable: false.\n` +
+          `${meta.listKey}.${meta.fieldKey} specifies defaultValue: { kind: 'autoincrement' } but doesn't specify db.isNullable: false.\n` +
             `Having nullable autoincrements on Prisma currently incorrectly creates a non-nullable column so it is not allowed.\n` +
             `https://github.com/prisma/prisma/issues/8663`
         )
       }
     }
-
-    const validation = {
-      isRequired: _validation?.isRequired ?? false,
-      min: _validation?.min ?? MIN_INT,
-      max: _validation?.max ?? MAX_INT,
+    if (min !== undefined && !Number.isInteger(min)) {
+      throw new Error(`${meta.listKey}.${meta.fieldKey} specifies validation.min: ${min} but it must be an integer`)
     }
-
-    for (const type of ['min', 'max'] as const) {
-      if (validation[type] > MAX_INT || validation[type] < MIN_INT) {
-        throw new Error(`The bigInt field at ${meta.listKey}.${meta.fieldKey} specifies validation.${type}: ${validation[type]} which is outside of the range of a 64bit signed integer(${MIN_INT}n - ${MAX_INT}n) which is not allowed`)
-      }
+    if (max !== undefined && !Number.isInteger(max)) {
+      throw new Error(`${meta.listKey}.${meta.fieldKey} specifies validation.max: ${max} but it must be an integer`)
+    }
+    if (min !== undefined && (min > MAX_INT || min < MIN_INT)) {
+      throw new Error(`${meta.listKey}.${meta.fieldKey} specifies validation.min: ${min} which is outside of the range of a 64-bit signed integer`)
+    }
+    if (max !== undefined && (max > MAX_INT || max < MIN_INT)) {
+      throw new Error(`${meta.listKey}.${meta.fieldKey} specifies validation.max: ${max} which is outside of the range of a 64-bit signed integer`)
     }
-    if (validation.min > validation.max) {
-      throw new Error(`The bigInt field at ${meta.listKey}.${meta.fieldKey} specifies a validation.max that is less than the validation.min, and therefore has no valid options`)
+    if (
+      min !== undefined &&
+      max !== undefined &&
+      min > max
+    ) {
+      throw new Error(`${meta.listKey}.${meta.fieldKey} specifies a validation.max that is less than the validation.min, and therefore has no valid options`)
     }
```

Comment:
I leveraged this opportunity to unify each of the numerical fields to share nearly the same code for `min` and `max` validation.